src/mdio/converters/segy.py

```diff
@@ -515,17 +505,22 @@ def segy_to_mdio(  # noqa PLR0913
         raise FileExistsError(err)
 
     segy_settings = SegyFileSettings(storage_options=input_path.storage_options)
-    segy_file = SegyFile(
-        url=input_path.as_posix(),
-        spec=segy_spec,
-        settings=segy_settings,
-        header_overrides=segy_header_overrides,
+    segy_kw: SegyFileArguments = {
+        "url": input_path.as_posix(),
+        "spec": segy_spec,
+        "settings": segy_settings,
+        "header_overrides": segy_header_overrides,
+    }
+    num_traces, sample_labels, segy_info = _read_segy_file_pp(segy_kw)
```

Comment:
let's combine all this info to a `SegyInfo` dataclass we can pass around instead of adding more function arguments like `num_traces` `sample_labels` etc. It would be a combination of SegyFileHeaderDump + new arguments we pass around.